### ISSUE TITLE
[Bug]: Setting API key in UI shows key in toast

### ISSUE THREAD

**Issue by rbren** [OpenDevin/OpenDevin#1477]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

When I add my API key via the settings modal, the toast shows my key in plaintext

### Current Version

```bash
On main as of `5f2d6f58f20a3f7db8e41d77bb86d1871381e367`
```

### Installation and Configuration

```bash
Running in dev mode
```

### Reproduction Steps

1. Open the UI
2. Set an API key
<img width="272" alt="Screenshot 2024-04-30 at 2 05 52 PM" src="https://github.com/OpenDevin/OpenDevin/assets/7611973/e3b328fa-3bb2-4b86-9726-366b9443b0c6">

**Comment by amodevinc** [comment]
Working on this

**Comment by amodevinc** [comment]
Resolved pending review in [PR#1480](https://github.com/OpenDevin/OpenDevin/pull/1480)

**Comment by iFurySt** [comment]
btw, i don't think store plaintext key to local storage is a good idea. a simple encryption is good 😶‍🌫️
